src/CommunityToolkit.Maui.Core/Essentials/FolderPicker/FolderPickerImplementation.android.cs

```diff
@@ -61,10 +64,25 @@ static string EnsurePhysicalPath(AndroidUri? uri)
 		const string uriSchemeFolder = "content";
 		if (uri.Scheme is not null && uri.Scheme.Equals(uriSchemeFolder, StringComparison.OrdinalIgnoreCase))
 		{
-			var split = uri.Path?.Split(':') ?? throw new FolderPickerException("Unable to resolve path.");
-			return $"{Android.OS.Environment.ExternalStorageDirectory}/{split[^1]}";
+			// Example path would be /tree/primary:DCIM, or /tree/SDCare:DCIM
```

Comment:
it seems like the code is the same as in  FileSaver. could you please extract it and reuse?

Reply:
Of course, where should I put the extracted code? Am I allowed to create a new namespace under Essentials? Like Essentials/Helpers or is there already a class / namespace? Thank you.

Reply:
Create it near this file: src/CommunityToolkit.Maui.Core/Essentials/AndroidRequestCode.android.cs

Reply:
Instead of a helper class I created an extension method that could be used globally to convert an Uri (Android) to a absolute path. Is that ok?